### ISSUE TITLE
[sklearn#34660] adding weights to the calculation for Ledoit Wolf

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#298]
**Mirrored from [scikit-learn#34660](https://github.com/scikit-learn/scikit-learn/issues/34660)**

Original title: adding weights to the calculation for Ledoit Wolf
Original labels: New Feature, Needs Triage
Created: 2026-08-06T01:53:06Z

---

> [!WARNING]
> This issue is not yet ready for a PR. If you are interested in contributing to scikit-learn, please have a look at our [contributing guidelines](https://scikit-learn.org/dev/developers/contributing.html), and in particular the sections for [new contributors](https://scikit-learn.org/dev/developers/contributing.html#new-contributors) and the ["Needs triage"](https://scikit-learn.org/dev/developers/contributing.html#issues-tagged-needs-triage) label.

### Describe the workflow you want to enable

Ledoit Wolf is a method for shrinking the empirical covariace matrix for data, and is already available in this library. The process essentially does three steps:
1) make the empirical covariance matrix
2) calculates some parameters (beta / delta) that are then used to calculate the shrinkage coefficient
3) use the results from steps (1) and (2) to make the desired output.

Given random p-vector X,  step (1) is just doing E[ X X^T ) - E[ X ] E[X^T ] the way you expect. Step (2) is calculating variables beta and delta that invariably involve calculating E[ X_i X_j ] and E[ X^2_i X^2_j ], just with a typical mean calculation. What I am proposing is extending these calculations, where weights can be leveraged to do the expected value calculations. This will allow the user to, for example, do the calculation on the exponentially weighted covariance matrix, but really use any weighting they choose. By setting weights = np.ones( X.shape[0] ) by default, we can maintain the current standard output. If desired, this procedure might possibly be extended to OAS (another shrinkage matrix).

### Describe your proposed solution

I am proposing adding the named variable "weights" to the "fit" function inputs for the LedoitWolf class (and any other necessary changes under the hood).

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to https://github.com/godofecht/flow-scikit/issues/125 in the flow-scikit repository. Further library-specific triage and implementation will happen there.
